### ISSUE TITLE
Gzip outgoing request body (POST|PATCH|PUT)?

### ISSUE THREAD

**Issue by kjs3** [request/request#2397]
[This issue](https://github.com/request/request/issues/1618) shows how to compress request bodies but I'm wondering why that wouldn't be done automatically with the `{gzip: true}` option?

Is this just a technical issue that hasn't been implemented yet?

**Comment by tzachshabtay** [comment]
Joining this question. I think the "gzip" option relates to the response compression, not to the request compression. It sounds to me like there should be 2 options: "gzip_response" and "gzip_request".

There's an issue from 2012 (https://github.com/request/request/issues/234) which was closed, the comments implying that this is now supported? But it's not, right? This left me confused.

**Comment by kjs3** [comment]
@tzachshabtay: Yes, this confused me too.

You are correct. The `gzip` option only refers to decompressing the response. I had to manually gzip the outgoing POST body.

I'm not sure this module should implement outgoing compression though. I don't believe servers are typically setup to inflate/decompress incoming request bodies. I controlled the server side in my case so I manually decompressed on that end. Saving some bandwidth seemed worth it but it was definitely a custom thing.

If I'm wrong and request body compression is expected on the server then I _would_ think this should be implemented.
